packages/core/src/lib/template-engine/built-in-extensions/query-builder/executorRunner.ts

```diff
@@ -1,19 +1,31 @@
-import { IDataQueryBuilder } from '@vulcan-sql/core/data-query';
 import {
+  DataResult,
   FilterRunner,
   FilterRunnerTransformOptions,
   VulcanInternalExtension,
 } from '@vulcan-sql/core/models';
+import { streamToArray } from '@vulcan-sql/core/utils';
 import { EXECUTE_FILTER_NAME } from './constants';
 
+const isDataResult = (response: any): response is DataResult => {
+  return response.getColumns && response.getData;
+};
+
 @VulcanInternalExtension()
 export class ExecutorRunner extends FilterRunner {
   public filterName = EXECUTE_FILTER_NAME;
 
   public async transform({
-    value,
+    value: builder,
   }: FilterRunnerTransformOptions): Promise<any> {
-    const builder: IDataQueryBuilder = value;
-    return builder.value();
+    const response = await builder.value();
+
+    // if input value is not a query builder, call the function .value and do nothing.
+    if (!isDataResult(response)) return response;
+
+    const { getData } = response;
+    const dataStream = getData();
+    const data = await streamToArray(dataStream);
```

Comment:
If we execute `builder.value()` and consume the stream to the array in the `ExecutorRunner` when play `RuntimeExtension` and it's the final result, then how could we convert the array result to json / csv format in the middleware?

Reply:
Okay, thanks